src/handlers/gestures/simpleGestures.ts

```diff
@@ -0,0 +1,429 @@
+  public config: Partial<CommonGestureConfig> = {};
+  public handlers: HandlerCallbacks = {
+    handlerTag: -1,
+  };
+
+  private addDependency(
+    key: 'simultaneousWith' | 'requireToFail',
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.config[key] = this.config[key]
+      ? [].concat(this.config[key], gesture)
+      : [gesture];
+  }
+
+  private toArray(x: unknown) {
+    return [].concat(x);
+  }
+
+  setRef(ref: React.RefObject<SimpleGesture>) {
+    this.config.ref = ref;
+    return this;
+  }
+
+  setOnBegan(callback: (event: HandlerStateChangeEventPayload) => void) {
+    this.handlers.onBegan = callback;
+    return this;
+  }
+
+  setOnStart(callback: (event: HandlerStateChangeEventPayload) => void) {
+    this.handlers.onStart = callback;
+    return this;
+  }
+
+  setOnEnd(
+    callback: (event: HandlerStateChangeEventPayload, success: boolean) => void
+  ) {
+    this.handlers.onEnd = callback;
+    return this;
+  }
+
+  setEnabled(enabled: boolean) {
+    this.config.enabled = enabled;
+    return this;
+  }
+
+  setMinPointers(minPointers: number) {
+    this.config.minPointers = minPointers;
+    return this;
+  }
+
+  setShouldCancelWhenOutside(value: boolean) {
+    this.config.shouldCancelWhenOutside = value;
+    return this;
+  }
+
+  setHitSlop(hitSlop: HitSlopType) {
+    this.config.hitSlop = hitSlop;
+    return this;
+  }
+
+  addSimultaneousGesture(
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.addDependency('simultaneousWith', gesture);
+
+    return this;
+  }
+
+  addRequiredToFailGesture(
+    gesture: SimpleGesture | React.RefObject<SimpleGesture>
+  ) {
+    this.addDependency('requireToFail', gesture);
+
+    return this;
+  }
+
+  simultaneousWith(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).simultaneousWith(other);
+  }
+
+  exclusiveWith(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).exclusiveWith(other);
+  }
+
+  requireToFail(other: SimpleGesture): GestureBuilder {
+    return new GestureBuilder(this).requireToFail(other);
+  }
+
+  initialize() {
+    this.handlerTag = getNextHandlerTag();
+    this.handlers = { ...this.handlers, handlerTag: this.handlerTag };
+
+    if (this.config.ref) {
+      this.config.ref.current = this;
+    }
+  }
+
+  build(): BuiltGesture {
+    const result = new BuiltGesture(this.prepare);
+    result.gestures = [this];
+    return result;
+  }
+
+  prepare = () => {
+    if (this.config.requireToFail !== undefined) {
+      this.config.requireToFail = this.toArray(this.config.requireToFail);
+    }
+
+    if (this.config.simultaneousWith !== undefined) {
+      this.config.simultaneousWith = this.toArray(this.config.simultaneousWith);
+    }
+  };
+}
+
+type TapGestureConfig = CommonGestureConfig & {
+  numberOfTaps: number;
+  maxDist: number;
+  maxDurationMs: number;
+  maxDelayMs: number;
+  maxDeltaX: number;
+  maxDeltaY: number;
+};
```

Comment:
I don't like inline props, if we in the future add one we'll have to change it in multiple places. Could we unify those?

Reply:
Done (I think) in 57f965641192e5efd43bd69a41b622c939d2745c